### PR TITLE
PGP form refacto

### PR DESCRIPTION
**Pull request by MonsieurLanza**
- Show spinner while generating key.
- Do not display verbose key details
- Allow user to save private key in a file.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## [0.18.2] 2019-04-26
 
+### Changed
```
Comment (iamdey) [inline CHANGELOG.md]: should be in unreleased part

File: src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      className, errors, onChange, options, ...props
+    } = this.props;
+    const id = `msfg-${uuidV1()}`;
+
+    return (
+      <FieldGroup className={classnames('m-select-field-group', className)} errors={errors}>
+        <select
+          id={id}
+          className="m-multi-select-field-group__select"
+          name="multiple"
```
Comment (iamdey) [inline src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx]: i'm not agree with `id` and `name` props, it should be dynamic and optional

File: src/frontend/web_application/src/scenes/UserSecurity/components/OpenPGPKeysDetails/presenter.jsx [hunk 3]
```diff
@@ -140,31 +155,18 @@ class OpenPGPKeysDetails extends Component {
...
-          </div>
+          <OpenPGPKeyForm
+            className="m-account-openpgp__form"
+            emails={user.contact.emails}
+            onImport={this.importKeys}
+            onGenerate={this.generateAndSaveKeys}
+            onImportFormChange={this.handleImportFormChange}
+            onGenerateFromChange={this.handleGenerateFormChange}
+            importForm={this.state.importForm}
+            generateForm={this.state.genereateForm}
```
Comment (iamdey) [inline src/frontend/web_application/src/scenes/UserSecurity/components/OpenPGPKeysDetails/presenter.jsx]: typo `genereateForm`

File: src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx [hunk 4]
```diff
@@ -0,0 +1,50 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { v1 as uuidV1 } from 'uuid';
+import classnames from 'classnames';
+import FieldGroup from '../FieldGroup';
+
+const propTypeOption = PropTypes.oneOfType([PropTypes.string, PropTypes.number]);
+const alphaNumPropType = PropTypes.oneOfType([PropTypes.string, PropTypes.number]);
+
+class MultiSelectFieldGroup extends PureComponent {
```
Comment (iamdey) [inline src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx]: it looks like it is unused

**Comment by iamdey** [comment]
unit tests doesn't pass

```
/srv/caliopen/src/frontend/web_application/src/scenes/UserSecurity/components/OpenPGPKey/presenter.jsx
    3:8   error  'classnames' is defined but never used                  no-unused-vars
    7:26  error  'TextareaFieldGroup' is defined but never used          no-unused-vars
    7:46  error  'DefList' is defined but never used                     no-unused-vars
  102:31  error  'publicKeyArmored' is assigned a value but never used   no-unused-vars
102:49 error 'privateKeyArmored' is assigned a value but never used no-unused-vars
```

File: src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx (hunk 4 above)
Comment (MonsieurLanza) [reply src/frontend/web_application/src/components/MultiSelectFieldGroup/index.jsx]: I removed references to it, a bit too quickly as it seems.

**Pull request merged**